### PR TITLE
Add url preview and open url handler

### PR DESCRIPTION
**Pull request by paul-nameless**

### CODE REVIEW

File: tg/controllers.py [hunk 1]
```diff
@@ -70,6 +70,32 @@ def __init__(self, model: Model, view: View, tg: Tdlib) -> None:
...
+            offset = entity["offset"]
+            length = entity["length"]
+            url = text[offset : offset + length]
+            urls.append(url)
+        if not urls:
+            self.present_error("No url to open")
+            return
+        if len(urls) == 1:
+            with suspend(self.view) as s:
+                s.call(f"open -a Firefox '{url}'")
```
Comment (lightme16) [inline tg/controllers.py:94]: - you should use `DEFAULT_OPEN` instead of `open`
- you shouldn't hardcode FF

File: tg/controllers.py [hunk 2]
```diff
@@ -70,6 +70,32 @@ def __init__(self, model: Model, view: View, tg: Tdlib) -> None:
...
+            return
+        with suspend(self.view) as s:
+            s.run_with_input("urlview", "\n".join(urls))
```
Comment (lightme16) [inline tg/controllers.py:97]: it should be added to the Readme as a requirement so use will know that it should install it to make this working

File: tg/views.py [hunk 3]
```diff
@@ -14,7 +14,6 @@
 
 MAX_KEYBINDING_LENGTH = 5
 MULTICHAR_KEYBINDINGS = (
-    "gg",
```
Comment (lightme16) [inline tg/views.py:17]: why was it deleted? I like to have `gg` shortcut. we need to take it back

File: tg/views.py [hunk 4]
```diff
@@ -304,11 +303,24 @@ def _format_reply_msg(
             msg = f"{reply_line}\n{msg}"
         return msg
 
+    def _format_url(self, msg_proxy: MsgProxy):
```
Comment (lightme16) [inline tg/views.py:306]: nice feature!

File: tg/views.py (hunk 3 above)
Comment (paul-nameless) [reply tg/views.py:17]: I've just thought it would be better to use g instead of gg (less uses g but vim uses gg).
Which one would you prefer?

**Comment by paul-nameless** [comment]
I'll merge the change cause I need to release package today.

**Pull request merged**